lib/common/busypoll.c

```diff
@@ -0,0 +1,444 @@
+#include "h2o.h"
+#include "h2o/busypoll.h"
+
+#include <errno.h>
+#include <error.h>
+
+#if defined(__linux__) && defined(SO_REUSEPORT) && defined(H2O_HAS_YNL_H)
+#include <linux/genetlink.h>
+#include <linux/netlink.h>
+#include <linux/filter.h>
+#include <linux/if.h>
+#include <sys/ioctl.h>
+#include <sys/sysinfo.h>
+#include <ynl/ynl.h>
+#include <ynl/netdev-user.h>
+
+static __thread int cpu_claimed = 0;
+static __thread const char *nic_claimed = NULL;
+static __thread unsigned int napi_id_claimed = 0;
+
+static void setup_queue(uint32_t cfg_ifindex, uint32_t cfg_defer_hard_irqs, uint64_t cfg_gro_flush_timeout,
```

Comment:
i think this probably needs to take a NAPI ID, as these options should be set on a per-NAPI basis... the implementation below seems to set them only on the first queue

Reply:
agree, see https://github.com/h2o/h2o/pull/3462#discussion_r1945900952